test/functional/actions/sendEvent.spec.js

```diff
@@ -383,3 +659,85 @@ test("shows error for data value that is more than one data element", async () =
+  datastreamsMocks.single,
+  datastreamMocks.withConfigOverrides
+)("populates overrides dropdowns with Blackbird config data", async () => {
+  await extensionViewController.init({
+    extensionSettings: {
+      instances: [
+        {
+          name: "alloy",
+          edgeConfigId: "aca8c786-4940-442f-ace5-7c4aba02118e",
+          sandbox: "prod"
+        }
+      ]
+    }
+  });
+
+  await overrideViewSelectors.envTabs.production.expectSelected();
+  await overrideViewSelectors.comboBoxes.eventDatasetOverride.expectExists();
+  await overrideViewSelectors.comboBoxes.eventDatasetOverride.openMenu();
+  await overrideViewSelectors.comboBoxes.eventDatasetOverride.expectMenuOptionLabels(
+    ["6335faf30f5a161c0b4b1444"]
+  );
+  await overrideViewSelectors.comboBoxes.eventDatasetOverride.selectMenuOption(
+    "6335faf30f5a161c0b4b1444"
+  );
+
+  await overrideViewSelectors.comboBoxes.idSyncContainerOverride.expectExists();
+  await overrideViewSelectors.comboBoxes.idSyncContainerOverride.openMenu();
+  await overrideViewSelectors.comboBoxes.idSyncContainerOverride.expectMenuOptionLabels(
+    ["107756", "107757"]
+  );
+  await overrideViewSelectors.comboBoxes.idSyncContainerOverride.selectMenuOption(
+    "107756"
+  );
+
+  await overrideViewSelectors.comboBoxes.targetPropertyTokenOverride.expectExists();
+  await overrideViewSelectors.comboBoxes.targetPropertyTokenOverride.openMenu();
+  await overrideViewSelectors.comboBoxes.targetPropertyTokenOverride.expectMenuOptionLabels(
+    [
+      "aba5431a-9f59-f816-7d73-8e40c8f4c4fd",
+      "65d186ff-be14-dfa0-75fa-546d93bebf91"
+    ]
+  );
+  await overrideViewSelectors.comboBoxes.targetPropertyTokenOverride.selectMenuOption(
+    "aba5431a-9f59-f816-7d73-8e40c8f4c4fd"
+  );
+
+  await overrideViewSelectors.comboBoxes.reportSuiteOverrides[0].expectExists();
+  await overrideViewSelectors.comboBoxes.reportSuiteOverrides[0].openMenu();
+  await overrideViewSelectors.comboBoxes.reportSuiteOverrides[0].expectMenuOptionLabels(
+    ["unifiedjsqeonly2", "unifiedjsqeonlylatest", "unifiedjsqeonlymobileweb"]
+  );
+  await overrideViewSelectors.comboBoxes.reportSuiteOverrides[0].selectMenuOption(
+    "unifiedjsqeonlylatest"
+  );
+
+  await extensionViewController.expectIsValid();
+});
+
+test("can type a datastream overrides", async () => {
+  await extensionViewController.init({
+    extensionSettings: {
+      instances: [
+        {
+          name: "alloy",
+          edgeConfigId: "aca8c786-4940-442f-ace5-7c4aba02118e",
+          sandbox: "prod"
+        }
+      ]
+    }
+  });
+
+  await overrideViewSelectors.envTabs.production.click();
+  await overrideViewSelectors.envTabs.production.expectSelected();
+  await overrideViewSelectors.datastreamInputMethod.freeform.click();
+  await overrideViewSelectors.datastreamInputMethod.freeform.expectChecked();
+  await overrideViewSelectors.datastreamIdFreeform.typeText("1234");
+
+  await extensionViewController.expectIsValid();
+});
```

Comment:
Can you add a test here that when choosing another instance, the datastream overrides switch to textfields and vice versa?
Also add a test that when you override the datastream, the drop downs populate with the allow lists of that datastream.